infrastructure/modules/platform/main.tf

```
@@ -97,3 +97,95 @@ resource "azurerm_application_insights" "app_insights" {
   resource_group_name = azurerm_resource_group.platform.name
   application_type    = "other"
 }
+
+resource "random_password" "password" {
+  length           = 32
+  special          = true
+  override_special = "_%@"
+}
+
+resource "random_string" "server" {
+  length  = 15
+  special = false
+}
+
+data "azurerm_client_config" "current" {
+}
+
+resource "azurerm_key_vault" "secret_vault" {
+  name                = format("%s%s", "kv-fnhs", random_string.server.result)
+  location            = var.location
+  resource_group_name = azurerm_resource_group.platform.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
+
+    key_permissions = [
+      "Create",
+      "Get",
+    ]
+
+    secret_permissions = [
+      "Set",
+      "Get",
+      "Delete",
+      "List",
+    ]
+  }
+
+  tags = {
+    environment = var.environment
+  }
+}
+
+resource "azurerm_key_vault_secret" "secret_synapse_password" {
+  name         = "synapse-password"
+  value        = random_password.password.result
+  key_vault_id = azurerm_key_vault.secret_vault.id
+
+  tags = {
+    environment = var.environment
+  }
+}
+
+resource "azurerm_mssql_server" "sql_server" {
+  name                         = "${azurerm_resource_group.platform.name}-synapse-sql-server"
+  resource_group_name          = azurerm_resource_group.platform.name
+  location                     = var.location
+  version                      = "12.0"
+  administrator_login          = "nhs-admin"
+  administrator_login_password = azurerm_key_vault_secret.secret_synapse_password.value
+  azuread_administrator {
+    login_username = var.ad_username
+    object_id      = var.ad_object_id
+  }
+
+  tags = {
+    environment = var.environment
+  }
+}
+
+resource "azurerm_sql_database" "sql_dw" {
```

I'm not used to the acronym dw for data warehouse
```suggestion
resource "azurerm_sql_database" "synapse_db" {
```
?